posawesome/posawesome/api/posapp.py

```diff
@@ -675,11 +676,18 @@ def delete_invoice(invoice):
 
 @frappe.whitelist()
 def get_items_details(pos_profile, items_data):
-    pos_profile = json.loads(pos_profile)
+    pos_profile = json.loads(pos_profile)    
     items_data = json.loads(items_data)
     warehouse = pos_profile.get("warehouse")
     result = []
 
+    cost_centers = frappe.get_all(
+            "Cost Center",
+            filters={"is_group": 0},
+            fields=["name as cost_center"],
```

Comment:
Need to use get_list instead of get_all to apply the permissions, and put a filter for the company